### PR TITLE
Add eslint

### PR DESCRIPTION
**Pull request by Jbithell**
[https://github.com/google/gts](https://github.com/google/gts)

## Version:

`v2`

## Checklist:

*Add any outstanding items to this checklist, but don't delete the default options*

- [ ] Updates documentation (if so - is it in this PR, or on the v2 branch?)
- [x] Tested locally

### CODE REVIEW

File: .github/workflows/reviewdog.yml [hunk 1]
```diff
@@ -30,4 +30,27 @@ jobs:
...
+    name: Lint App
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: eviewdog/action-eslint@v1
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          reporter: github-pr-check
+          workdir: "./app"
+          eslint_flags: "\"{src,apps,libs,test}/**/*.ts\""
```
Comment (Robert-Watts) [inline .github/workflows/reviewdog.yml:45]: will this include tsx?

File: .github/workflows/reviewdog.yml [hunk 2]
```diff
@@ -30,4 +30,27 @@ jobs:
...
+    name: Lint Api
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: eviewdog/action-eslint@v1
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          reporter: github-pr-check
+          workdir: "./api"
+          eslint_flags: "\"{src,apps,libs,test}/**/*.ts\""
```
Comment (Robert-Watts) [inline .github/workflows/reviewdog.yml:56]: same again

File: .github/workflows/reviewdog.yml (hunk 1 above)
Comment (Jbithell) [reply .github/workflows/reviewdog.yml:45]: I suspect it won't include anything and will just fail miserably

File: app/tsconfig.json [hunk 3]
```diff
@@ -1,26 +1,11 @@
...
-    "resolveJsonModule": true,
-    "isolatedModules": true,
-    "noEmit": true,
-    "jsx": "react-jsx"
+    "rootDir": ".",
+    "outDir": "build"
   },
   "include": [
-    "src"
+    "src/**/*.ts",
```
Comment (Robert-Watts) [inline app/tsconfig.json:8]: tsx

Comment (Jbithell) [reply app/tsconfig.json:8]: Good spot thx

**Pull request merged**